pipelines/google_sheets/data_validator.py

```diff
@@ -0,0 +1,80 @@
+# This is a helper module that contains function which validate data
+
+from typing import Union
+from dateutil import parser
+from datetime import datetime
+
+import pytest
+
+
+# this function takes an url to a Google spreadsheet and computes the spreadsheet id from it. returns an empty string if the url format is not correct
+# spreadsheet url formula: https://docs.google.com/spreadsheets/d/<spreadsheet_id>/edit
+# @input: url- str
+# @output spreadsheet_id: str
+def process_url(url: str) -> str:
+
+    # this string comes before the id
+    url_id_identifier = "d"
+    # split on the '/'
+    parts = url.split("/")
```

Comment:
do not write your own parser. reuse what python has
1. parse the url with https://docs.python.org/3/library/urllib.parse.html
2. get the `path` part
3. use regular expression to get component after `spreadsheets/d/`

did you use chatGPT to generate code? it can actually generate the code you need. my advice is to use it to get a first version of a code that seems like a common problem
![image](https://user-images.githubusercontent.com/17202864/212306120-943ffd2b-7383-4f18-b4d4-9f011b6ec5a8.png)